### PR TITLE
[echarts | zrender] textStyle.rich & label.color can be a function ...

### PR DESCRIPTION
**Pull request by gcdxuzhiwei**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/echarts/index.d.ts [hunk 1]
```diff
@@ -1582,7 +1582,8 @@ declare namespace echarts {
                 /**
                  * @todo describe
                  */
-                interface Label extends TextStyleWithRich {
+                interface Label extends Omit<TextStyleWithRich,'color'> {
```
Comment (amcasey) [inline types/echarts/index.d.ts:1585]: Does `TextStyleWithRich` have a `color` property to omit?

File: types/echarts/index.d.ts [hunk 2]
```diff
@@ -1582,7 +1582,8 @@ declare namespace echarts {
...
+                    color?: string | ((val:string) => EChartOption.Color) | undefined;
```
Comment (amcasey) [inline types/echarts/index.d.ts:1586]: I would find a test showing an example usage helpful.

File: types/zrender/index.d.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ declare namespace zrender {
     type Y = number;
     type X2 = number;
     type Y2 = number;
-    type GlobalCoords = boolean;
+    type Global = boolean;
```
Comment (amcasey) [inline types/zrender/index.d.ts:12]: This looks correct, but unrelated.  Next time, can you please open a separate PR?

**Review by amcasey — CHANGES_REQUESTED** [review]
Some minor questions

File: types/echarts/index.d.ts (hunk 1 above)
Comment (gcdxuzhiwei) [reply types/echarts/index.d.ts:1585]: interface TextStyleWithRich extends TextStyle，interface TextStyle extends BaseTextStyle，interface BaseTextStyle has a **color** property

File: types/zrender/index.d.ts (hunk 3 above)
Comment (gcdxuzhiwei) [reply types/zrender/index.d.ts]: this change has been reverted

File: types/echarts/index.d.ts (hunk 2 above)
Comment (gcdxuzhiwei) [reply types/echarts/index.d.ts:1586]: image

File: types/echarts/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/echarts/index.d.ts:1585]: Oops, I was looking at `BaseTextStyleWithRich` instead of `TextStyleWithRich`.

**Review by amcasey — APPROVED** [review]
Thanks for the explanations!

**Review by amcasey — APPROVED** [review]

**Comment by gcdxuzhiwei** [comment]
Ready to merge

**Pull request merged**